container/lxd/network.go

```diff
@@ -49,15 +54,23 @@ func (s *Server) EnsureIPv4(netName string) (bool, error) {
 	return modified, nil
 }
 
-func (s *Server) VerifyDefaultBridge(profile *api.Profile, eTag string) error {
+// VerifyNetworkDevice attempts to ensure that there is a network usable by LXD
+// and that there is an eth0 NIC device with said network as its parent.
+// If we are not operating as part of a cluster, an attempt will be made to
+// create a missing eth0 device with the default LXD bridge as its parent.
+// If in clustered mode this attempt is not made - the onus of network
+// configuration falls on the operator.
+// NOTE (manadart): Cluster behaviour is subject to change.
+func (s *Server) VerifyNetworkDevice(profile *api.Profile, eTag string) error {
 	eth0, ok := profile.Devices["eth0"]
 	if !ok {
 		// On LXD >= 2.3 there is no bridge config by default.
-		// Ensure that the network bridge and eth0 device both exist.
-		if s.networkAPISupport {
+		// If not part of a cluster, ensure that the network bridge and eth0
+		// device both exist.
+		if s.networkAPISupport && !s.clustered {
```

Comment:
I'm very hesitant to add "if clustered" checks if we can help it. If it would be possible to just support networks rather than forcibly defining that we have exactly an 'eth0' on the default bridge.
What configuration do we get out of the box if we don't force eth0? Do we just have no networking on containers by default?

I'm just concerned that we're doing a lot of bending over backward to maintain current behavior when we don't really *want* current behavior.

Reply:
so after talking it over with you, I'm less worried about s.clustered because it isn't changing the *verify* step to know if we expect containers to work, it is just changing the "go modify the configuration" because we don't expect it to work as well.